src/actions/WalletListMenuActions.tsx

```diff
@@ -139,7 +140,56 @@ export function walletListMenuAction(
         dispatch(showSplitWalletModal(walletId, option.replace('split', '')))
       }
     }
+    case 'viewPrivateViewKey': {
+      return (dispatch, getState) => {
+        const state = getState()
+        const { currencyWallets } = state.core.account
+        const { displayPublicSeed, currencyInfo } = currencyWallets[walletId]
+        const { xpubExplorer } = currencyInfo
+
+        const copy: ButtonInfo = {
+          label: lstrings.fragment_request_copy_title,
+          type: 'secondary'
+        }
+        const link: ButtonInfo = {
+          label: lstrings.transaction_details_show_advanced_block_explorer,
+          type: 'secondary'
+        }
+        const buttons = xpubExplorer != null ? { copy, link } : { copy }
 
+        Airship.show<'copy' | 'link' | undefined>(bridge => (
+          <ButtonsModal
+            bridge={bridge}
+            buttons={buttons as { copy: ButtonInfo; link: ButtonInfo }}
+            closeArrow
+            message={displayPublicSeed ?? ''}
+            title={lstrings.fragment_wallets_view_private_view_key}
+          >
+            <Alert
+              type="warning"
+              title={lstrings.string_warning}
+              marginRem={0.5}
+              message={lstrings.fragment_wallets_view_private_view_key_warning}
+              numberOfLines={0}
+            />
```

Comment:
Aside from this one addition, and changing `lstrings.fragment_wallets_view_xpub` to `lstrings.fragment_wallets_view_private_view_key`, this whole switch case is identical to the following one. Can we combine them somehow?

```ts
case 'viewPrivateViewKey':
case 'viewXPub': {
  // ... the same stuff ...
  switchString === 'viewPrivateViewKey' ? lstrings.fragment_wallets_view_private_view_key : lstrings.fragment_wallets_view_xpub
  // ... more similar stuff...
}
```

Reply:
I can combine them, but in my mind these are different cases altogether and the code duplication is correct. If there are separate cases, then we should handle the cases separately. But, I'll try the change you suggest.